src/curve_fit.jl

```diff
@@ -88,3 +97,22 @@ function estimate_errors(fit::LsqFitResult, alpha=0.95)
 	dist = TDist(fit.dof)
 	std_error *= quantile(dist, alpha)
 end
+
+function estimate_errors(fit::LsqFitResult, vars::Vector{Float64}, alpha=0.95; atol=1e-10)
+    # computes (1-alpha) error estimates from
+    #   fit   : a LsqFitResult from a curve_fit()
+    #   alpha : alpha percent confidence interval, (e.g. alpha=0.95 for 95% CI)
+    covar = estimate_covar(fit, vars)
+    
+    # then the standard errors are given by the sqrt of the diagonal
+    vars = diag(covar)
+    vratio = minimum(vars)/maximum(vars) 
+    if !isapprox(vratio, 0.0, atol=atol) && vratio < 0.0
+        error("Covariance matrix is negative")
+    end
+    std_error = sqrt(abs(vars))
```

Comment:
I think up to numerical precision it is possible it is negative, as long as it is approximately zero — mathematically, of course, the construction should ensure positive semidefinites.

In my tests with the Ramsey data negativity does seem to come up, if I remember correctly, but adding a tolerance around 0.0 (as implied by the `isapprox` test) deals with that.